src/main/java/com/flipkart/ranger/healthservice/HealthService.java

```diff
@@ -0,0 +1,54 @@
+package com.flipkart.ranger.healthservice;
+
+import com.flipkart.ranger.healthcheck.HealthcheckStatus;
+import com.flipkart.ranger.healthservice.monitor.IsolatedHealthMonitor;
+import com.flipkart.ranger.healthservice.monitor.Monitor;
+
+/**
+ * An interface which can be used to track the health of any service
+ * You can add a set of monitors, and a general health will be tracked by
+ *
+ * @see ServiceHealthAggregator
+ */
+public interface HealthService {
+
+    /**
+     * Register a monitor to the service, to setup a continuous monitoring on the monitor
+     * <p/>
+     * this method can be used to add a {@link IsolatedHealthMonitor} which will later be
+     * scheduled at regular intervals and monitored to generate and maintain an aggregated health of the service
+     * the scheduling will happen in an isolated thread
+     *
+     * @param monitor an implementation of the {@link IsolatedHealthMonitor}
+     */
+    void addIsolatedMonitor(IsolatedHealthMonitor monitor);
```

Comment:
why parameter is not of interface type like the other monitor? any specific reason? both sounds same so can be one method.